### PR TITLE
refactor: migrate CaptchaFactory to Typescript

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR is a pure refactor of CaptchaFactory to Typescript.

Related to #144

### CODE REVIEW

File: src/config/feature-manager/util/FeatureManager.class.ts [hunk 1]
```diff
@@ -109,11 +109,11 @@ export default class FeatureManager {
    * and whether requested feature is enabled
    * @param name the name of the feature to return
    */
-  get(name: FeatureNames): RegisteredFeature<FeatureNames> {
+  get<K extends FeatureNames>(name: K): RegisteredFeature<K> {
     return {
       isEnabled: this.isEnabled(name),
       // TODO (#317): remove usage of non-null assertion
```
Comment (arshadali172) [inline src/config/feature-manager/util/FeatureManager.class.ts:115]: remove comment

File: src/config/feature-manager/types.ts [hunk 2]
```diff
@@ -80,6 +80,11 @@ export interface IFeatureManager {
   [FeatureNames.WebhookVerifiedContent]: IWebhookVerifiedContent
 }
 
+export interface RegisteredFeature<T extends FeatureNames> {
+  isEnabled: boolean
+  props?: IFeatureManager[T]
```
Comment (arshadali172) [inline src/config/feature-manager/types.ts:85]: why is props optional? Isn't it always defined for all features?

File: src/app/factories/captcha.factory.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { RequestHandler } from 'express'
+
+import FeatureManager, {
+  FeatureNames,
+  RegisteredFeature,
+} from '../../config/feature-manager'
+import { captchaCheck } from '../controllers/submissions.server.controller'
+
+// TODO(#144): Migrate middlewares and request handlers to the controller layer
+interface ICaptchaFactory {
```
Comment (arshadali172) [inline src/app/factories/captcha.factory.ts:11]: Can we move this to the `/types` folder

**Review by arshadali172 — CHANGES_REQUESTED** [review]
minor changes suggested

File: src/config/feature-manager/types.ts (hunk 2 above)
Comment (karrui) [reply src/config/feature-manager/types.ts:85]: It's only defined if the feature is enabled, so there is still a chance it is undefined

File: src/app/factories/captcha.factory.ts (hunk 3 above)
Comment (karrui) [reply src/app/factories/captcha.factory.ts:11]: Same reasoning I gave in the SmsFactory refactor; I use `typeof` to get the types leh, I scared circular dependency :X

File: src/config/feature-manager/types.ts (hunk 2 above)
Comment (arshadali172) [reply src/config/feature-manager/types.ts:85]: you are right.

File: src/app/factories/captcha.factory.ts (hunk 3 above)
Comment (arshadali172) [reply src/app/factories/captcha.factory.ts:11]: alright

**Review by arshadali172 — APPROVED** [review]

**Pull request merged**